src/ReflectionEngine.php

```diff
@@ -81,6 +81,9 @@ public static function init(LocatorInterface $locator)
     public static function setMaximumCachedFiles($newLimit)
     {
         self::$maximumCachedFiles = $newLimit;
+        if (count(self::$parsedFiles) > $newLimit) {
+            self::$parsedFiles = array_slice(self::$parsedFiles, 0, $newLimit);
```

Comment:
This would keep oldest entries in array. But when parsing new file oldest entry gets removed.

Maybe we should be slicing from right array side to be consistent?